### PR TITLE
Onboarding advisor for autogeneration

### PR DESCRIPTION
**Pull request by leni-msft**
```
> azure-schema-generator@1.0.0 generate-single /home/leni/workspace/azure-resource-manager-schemas/generator
> ts-node cmd/generatesingle "advisor/resource-manager"

[/tmp/schm_azspc] executing: git fsck
[/tmp/schm_azspc] executing: git reset -q .
[/tmp/schm_azspc] executing: git checkout -q -- .
[/tmp/schm_azspc] executing: git clean -q -fd
[/tmp/schm_azspc] executing: git gc
[/tmp/schm_azspc] executing: git fetch -q
[/tmp/schm_azspc] executing: git checkout -q origin/master
Using autogenlist config:
{
  "basePath": "advisor/resource-manager",
  "namespace": "Microsoft.Advisor",
  "resourceConfig": [
    {
      "type": "recommendations/suppressions",
      "scopes": 32
    }
  ]
}
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/ifjgbw1z0v --tag=all-api-versions --api-version=2016-07-12-preview --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/advisor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[6.4 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2016-07-12-preview/Microsoft.Advisor.json
Provider Namespace: Microsoft.Advisor
API Version: 2016-07-12-preview
Resource Types (Extension Scope):
- recommendations/suppressions
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/q6cseirv0bc --tag=all-api-versions --api-version=2017-03-31 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/advisor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[4.3 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-03-31/Microsoft.Advisor.json
Provider Namespace: Microsoft.Advisor
API Version: 2017-03-31
Resource Types (Extension Scope):
- recommendations/suppressions
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/1yqdu09h8gc --tag=all-api-versions --api-version=2017-04-19 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/advisor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[5.1 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2017-04-19/Microsoft.Advisor.json
Provider Namespace: Microsoft.Advisor
API Version: 2017-04-19
Resource Types (Subscription Scope):
- configurations
Resource Types (Resource Group Scope):
- configurations
Resource Types (Extension Scope):
- recommendations/suppressions
================================================================================================================================
[/home/leni/workspace/azure-resource-manager-schemas/generator] executing: autorest-beta --version=3.0.6274 --use=@autorest/azureresourceschema@3.0.79 --azureresourceschema --output-folder=/tmp/shcv3b0npp --tag=all-api-versions --api-version=2020-01-01 --title=none --pass-thru:subset-reducer /tmp/schm_azspc/specification/advisor/resource-manager/readme.md
AutoRest code generation utility [cli version: 3.0.6187; node: v13.9.0, max-memory: 8192 gb]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
   Loading AutoRest core      '/home/leni/.autorest/@autorest_core@3.0.6274/node_modules/@autorest/core/dist' (3.0.6274)
   Loading AutoRest extension '@autorest/azureresourceschema' (3.0.79->3.0.79)
[4.82 s] Generation Complete
================================================================================================================================
Filename: /home/leni/workspace/azure-resource-manager-schemas/schemas/2020-01-01/Microsoft.Advisor.json
Provider Namespace: Microsoft.Advisor
API Version: 2020-01-01
Resource Types (Subscription Scope):
- configurations
Resource Types (Resource Group Scope):
- configurations
Resource Types (Extension Scope):
- recommendations/suppressions
================================================================================================================================
```

### CODE REVIEW

File: generator/whitelist.ts [hunk 1]
```diff
@@ -4,6 +4,16 @@ import { lowerCaseEquals } from './utils';
 
 // Run "npm run list-basepaths" to discover all the valid readme files to add to this list
 const whitelist: WhitelistConfig[] = [
+    {
+        basePath: 'advisor/resource-manager',
+        namespace: 'Microsoft.Advisor',
+        resourceConfig: [
+            {
+                type: 'recommendations/suppressions',
+                scopes: ScopeType.Tenant | ScopeType.Subcription | ScopeType.ResourceGroup | ScopeType.ManagementGroup,
```
Comment (anthony-c-martin) [inline generator/whitelist.ts]: Is this correct? From the swagger, doesn't look like this resource is deployable at tenant, subscription or Rg scopes.

Comment (leni-msft) [reply generator/whitelist.ts]: The path in [2020-01-01/advisor.json#L505](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/advisor/resource-manager/Microsoft.Advisor/stable/2020-01-01/advisor.json#L505) is `"/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}":` 
Can I ask how to tell if it's deployable at a scope or not?

Comment (anthony-c-martin) [reply generator/whitelist.ts]: There's no easy way to figure this out. Looking at the docs [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/extension-resource-types) and [here](https://docs.microsoft.com/en-us/rest/api/advisor/suppressions/create) it does appear this is an extension resource. Please could you mark this as `ScopeType.Extension` and re-run the autogeneration?

Comment (apclouds) [reply generator/whitelist.ts:17]: @leni-msft any updates on this?

Comment (leni-msft) [reply generator/whitelist.ts]: resolved. pls check again.

**Pull request closed without merging**